compiler/src/iree/compiler/Codegen/Common/TransformExtensions/CommonExtensions.cpp

```diff
@@ -152,8 +153,24 @@ static LogicalResult gpuComprehensiveBufferizeCopyFn(OpBuilder &builder,
   return success();
 }
 
+void transform_dialect::IREEBufferizeOp::getEffects(
+    SmallVectorImpl<MemoryEffects::EffectInstance> &effects) {
+  transform::consumesHandle(getTarget(), effects);
+  transform::producesHandle(getResult(), effects);
+  transform::modifiesPayload(effects);
+}
+
 DiagnosedSilenceableFailure transform_dialect::IREEBufferizeOp::apply(
     transform::TransformResults &results, transform::TransformState &state) {
+  ArrayRef<Operation *> payload = state.getPayloadOps(getTarget());
+  if (payload.size() != 1 ||
+      !isa<HAL::ExecutableOp, HAL::ExecutableVariantOp>(payload.front())) {
+    state.getTopLevel()->emitOpError(
+        "requires exactly a single HAL::ExecutableOp or "
+        "HAL::ExecutableVariantOp target op.");
+    return DiagnosedSilenceableFailure(
+        reportUnknownTransformError(payload.front()));
```

Comment:
This looks fishy. It emits an error, and then returns a difference silenceable error. Even if the parent wants to silence the result, the first error is still emitted...

Reply:
Thanks, fixed